Add public link for setlist lyrics

johnjwang opened the pull request:
Requesting comments.

The URL format is:
`/setlists/lyrics/1?key=d76fc0514272763410602e0d8ecf4744`

It's not pretty, but does anyone have any objections? This gets around the fact that personal and group setlists have different URL structures: `/:group/:setlist/lyrics` is possible but `/personal/:setlist/lyrics` is not unique.

johnjwang commented on views/setlists/view.php:
```diff
@@ -59,8 +59,14 @@
-  <a href="<?= $base_url ?>/songs" data-role="button" data-theme="b" <?php if (count($songs) == 0) echo 'disabled' ?>>View All</a>
+  <a href="<?= $base_url ?>/songs" data-role="button" data-theme="b" <?php if (count($songs) == 0) echo 'disabled' ?>>View Chords</a>
+<?php
+  # Not sure if this is best place to do this
```
So I had a question about this. I need to generate access_key for old songsets at least before they're viewed. However, is it better to generate access_key for ALL old songsets in migration? or in a maintenance script?

chrisjbaik commented:
Oh I see what you mean regarding authentication. That makes sense.

If the migration can do it, that seems favorable. If not, a script can do
haha

On Sat, Jan 28, 2017, 2:24 PM johnjwang <notifications@github.com> wrote:

> *@johnjwang* commented on this pull request.
>
> I think I'll use /setlists/:id, however I can't nest it under the group
> because the group requires an authenticated user.
> ------------------------------
>
> In views/setlists/view.php
> <https://github.com/chrisjbaik/psalted/pull/42#pullrequestreview-18979420>
> :
>
> > @@ -59,8 +59,14 @@
>      </div>
>    </div>
>
> -  <a href="<?= $base_url ?>/songs" data-role="button" data-theme="b" <?php if (count($songs) == 0) echo 'disabled' ?>>View All</a>
> +  <a href="<?= $base_url ?>/songs" data-role="button" data-theme="b" <?php if (count($songs) == 0) echo 'disabled' ?>>View Chords</a>
> +<?php
> +  # Not sure if this is best place to do this
>
> So I had a question about this. I need to generate access_key for old
> songsets at least before they're viewed. However, is it better to generate
> access_key for ALL old songsets in migration? or in a maintenance script?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/chrisjbaik/psalted/pull/42#pullrequestreview-18979420>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAu4ZqNILo0ugkBAlF7tPJRwKuz93PwSks5rW5X2gaJpZM4LaHTo>
> .
>
-- 
- Chris

johnjwang commented:
Finally got around to addressing the reviews! Also did some visual cleanup of the lyrics view so it looks like the existing views.

![screenshot from 2017-02-26 20-41-10](https://cloud.githubusercontent.com/assets/90739/23345985/f185b634-fc63-11e6-8f2e-45a97615a314.png)

chrisjbaik commented:
@johnjwang sick thanks!
